subprojects/gradle/lint-report/src/main/kotlin/com/avito/android/lint/LintSlackAlert.kt

```diff
@@ -0,0 +1,65 @@
+package com.avito.android.lint
+
+import com.avito.slack.SlackClient
+import com.avito.slack.model.SlackChannel
+import com.avito.utils.logging.CILogger
+
+interface LintSlackAlert {
+
+    fun alert(
+        models: List<LintReportModel>,
+        channel: SlackChannel
+    )
+
+    class Impl(
+        private val slackClient: SlackClient,
+        private val logger: CILogger
+    ) : LintSlackAlert {
+
+        private val tag = "LintSlackAlert"
+
+        override fun alert(
+            models: List<LintReportModel>,
+            channel: SlackChannel
+        ) {
+            logger.debug("$tag: Sending lint alert...")
+
+            models.forEach { model ->
+                if (shouldSendAlert(model)) {
+                    slackClient.uploadHtml(
+                        channel = channel,
+                        message = buildSlackMessage(model),
+                        file = model.htmlFile
+                    ).fold(
+                        { logger.debug("$tag: Report sent successfully") },
+                        { error -> logger.critical("$tag: Can't send report ${error.message}") }
+                    )
+                } else {
+                    logger.debug("$tag Skip sending lint alert")
+                }
+            }
+        }
+
+        private fun shouldSendAlert(model: LintReportModel): Boolean {
+            return (model is LintReportModel.Valid) && model.issues.any { it.severity == LintIssue.Severity.ERROR }
+        }
+
+        private fun buildSlackMessage(model: LintReportModel): String {
+            return buildString {
+                appendln("*Critical lint problems detected in develop!*")
```

Comment:
Why does this abstraction know about branches?

Reply:
it should not, you're right